waiter/integration/waiter/health_check_test.clj

```diff
@@ -0,0 +1,158 @@
+;;
+;; Copyright (c) Two Sigma Open Source, LLC
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;  http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+(ns waiter.health-check-test
+  (:require [clojure.data.json :as json]
+            [clojure.test :refer :all]
+            [clojure.walk :as walk]
+            [waiter.util.client-tools :refer :all]
+            [waiter.util.http-utils :as hu]))
+
+(defn assert-ping-response
+  [health-check-protocol idle-timeout service-id response]
+  (let [{:keys [ping-response service-state]}
+        (-> (some-> response :body json/read-str)
+          (update-in ["ping-response" "body"] #(some-> % json/read-str))
+          walk/keywordize-keys)]
+    (assert-response-status response 200)
+    (if (nil? idle-timeout)
+      (do
+        (is (= "received-response" (get ping-response :result)) (str ping-response))
+        (is (= (hu/backend-protocol->http-version health-check-protocol)
+               (get-in ping-response [:body :protocol-version]))
+            (str ping-response))
+        (is (= "get" (get-in ping-response [:body :request-method])) (str ping-response))
+        (is (= {:exists? true :healthy? true :service-id service-id :status "Running"} service-state)))
+      (do
+        (is (= "timed-out" (get ping-response :result)) (str ping-response))
+        (is (= {:exists? true :healthy? false :service-id service-id :status "Starting"} service-state))))))
```

Comment:
Is the status here guaranteed to be `Starting`? Or could it be `Running` in some situations?

Reply:
Yes, because the request will timeout before the service has a healthy instance.